### PR TITLE
[node] Update typings to v22.5.0

### PR DESCRIPTION
**Pull request by Semigradsky**
https://github.com/nodejs/node/releases/tag/v22.5.0

- http: expose websockets
- lib: add `node:sqlite` module
- path: add `matchesGlob` method
- worker: add `postMessageToThread`
- process: port on-exit-leak-free to core

------------------------------------

Fixes https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/70210

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/node/sqlite.d.ts
1. types/node/process.d.ts
1. types/node/worker_threads.d.ts
1. types/node/scripts/generate-docs/docs.ts
1. types/node/scripts/generate-docs/ast-processing.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e4eac4fd454e6929ea9247a49e5da862de2f0d38
</p>

**Comment by Semigradsky** [comment]
@Renegade334 could you please review it?

File: types/node/sqlite.d.ts [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+         * the values in `namedParameters` and `anonymousParameters`.
+         * @since v22.5.0
+         * @param namedParameters An optional object used to bind named parameters. The keys of this object are used to configure the mapping.
+         * @param anonymousParameters Zero or more values to bind to anonymous parameters.
+         * @return An array of objects. Each object corresponds to a row returned by executing the prepared statement. The keys and values of each object correspond to the column names and values of
+         * the row.
+         */
+        all(...anonymousParameters: SupportedValueType[]): unknown[];
+        all(namedParameters: Record<string, SupportedValueType>, ...anonymousParameters: SupportedValueType[]): unknown[];
+        all(firstParameter: Record<string, SupportedValueType> | SupportedValueType, ...anonymousParameters: SupportedValueType[]): unknown[];
```
[the remark this answers is not in the document]
Comment (Semigradsky) [reply types/node/sqlite.d.ts]: Yeah, looks like it works with two signatures. I thought that a "common" signature was needed.

**Review by Renegade334 — COMMENTED** [review]
LGTM, although there's an issue with the v22.4 webstorage declarations (#70320) that should probably be resolved before the version is bumped.

**Comment by Semigradsky** [comment]
@jakebailey please take a look!

**Comment by jakebailey** [comment]
I assume this is still waiting on #70328 but it's likely fine once that's in.

**Review by jakebailey — APPROVED** [review]

**Comment by Semigradsky** [comment]
Ready to merge

**Pull request merged**